gobblin-data-management/src/main/java/gobblin/data/management/retention/DatasetRetentionRateLimiter.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.data.management.retention;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import gobblin.runtime.Limiter;
+import gobblin.runtime.RateBasedLimiter;
+
+/***
+ * Singleton class for controlling the hdfs call rate during datasets cleaning.
+ */
+public class DatasetRetentionRateLimiter {
+
+  private final Limiter rateLimiter;
+
+  private static DatasetRetentionRateLimiter INSTANCE = null;
+
+  private static final Logger LOG = LoggerFactory.getLogger(DatasetRetentionRateLimiter.class);
+ 
+  // Default rate limit.
+  private static long rateLimit = 1000;
+
+  private DatasetRetentionRateLimiter() {
+    this.rateLimiter = new RateBasedLimiter(getRateLimit());
+    this.rateLimiter.start();
+  }
+
+  public static synchronized DatasetRetentionRateLimiter getInstance() {
+    if (DatasetRetentionRateLimiter.INSTANCE == null) {
+      DatasetRetentionRateLimiter.INSTANCE = new DatasetRetentionRateLimiter();
+    }
+    return DatasetRetentionRateLimiter.INSTANCE;
+  }
+
+  public static long getRateLimit() {
+    return rateLimit;
+  }
+  
+  public static synchronized void setRateLimit(long rate) {
+    LOG.info("Rate limit is set to: " + rateLimit);
+    rateLimit = rate;
+  }
+  
+  public Limiter getRateLimiter() {
+    return this.rateLimiter;
+  }
+
+  public void acquirePermits(long permits) {
```

Comment:
You can create a convenience method `public void acquirePermit()` for acquiring a single permit since it's used a lot.